dev/Repeater/FlexboxLayout.cpp

```diff
@@ -0,0 +1,490 @@
+    {
+        sorted.push_back(child);
+    }
+    std::sort(sorted.begin(), sorted.end(), [](winrt::UIElement const& a, winrt::UIElement const& b)
+        {
+            return (GetOrder(a) - GetOrder(b));
+        });
+
+    return sorted;
+}
+
+void FlexboxLayout::InitializeForContextCore(winrt::LayoutContext const& context)
+{
+    auto state = context.LayoutState();
+    winrt::com_ptr<FlexboxLayoutState> flexboxState = nullptr;
+    if (state)
+    {
+        flexboxState = state.as<FlexboxLayoutState>();
+    }
+
+    if (!flexboxState)
+    {
+        if (state)
+        {
+            throw winrt::hresult_error(E_FAIL, L"LayoutState must derive from FlexboxLayoutState.");
+        }
+
+        flexboxState = winrt::make_self<FlexboxLayoutState>();
+    }
+
+    context.LayoutStateCore(*flexboxState);
+}
+
+void FlexboxLayout::UninitializeForContextCore(winrt::LayoutContext const& context)
+{
+}
+
+winrt::Size FlexboxLayout::MeasureOverride(
+    winrt::NonVirtualizingLayoutContext const& context,
+    winrt::Size const& availableSize)
+{
+    auto state = context.LayoutState().as<FlexboxLayoutState>();
+    state->Rows.clear();
+
+    unsigned int itemsInRow = 0;
+    float growInRow = 0.0;
+    //shrink
+    float shrinkInRow = 0.0;
+
+    float usedInCurrentMainAxis = 0;
+    float usedInCurrentCrossAxis = 0;
+
+    float usedMainAxis = 0;
+    float usedCrossAxis = 0;
+    float basisInRow = 0;
+
+    auto completeRow = [&]()
+    {
+        FlexboxLayoutState::RowMeasureInfo newRow;
+        newRow.MainAxis = usedInCurrentMainAxis;
+        newRow.CrossAxis = usedInCurrentCrossAxis;
+        newRow.Count = itemsInRow;
+        newRow.Grow = growInRow;
+        newRow.Shrink = shrinkInRow;
+        //basis, do we need to save this inRow?
+        newRow.Basis = basisInRow;
+        state->Rows.emplace_back(newRow);
+
+        itemsInRow = 0;
+        growInRow = 0.0;
+        usedMainAxis = std::max(usedMainAxis, usedInCurrentMainAxis);
+        usedInCurrentMainAxis = 0;
+        usedCrossAxis += usedInCurrentCrossAxis;
+        usedInCurrentCrossAxis = 0;
+
+
+
+    };
+   
+    
+
+    std::vector<winrt::UIElement> sortedChildren = ChildrenSortedByOrder(context.try_as<winrt::NonVirtualizingLayoutContext>());
+    for (winrt::UIElement const& child : sortedChildren)
+    {
+        
+
+        // Give each child the maximum available space
+        // TODO: What about flex-shrink? Should we try them with less?
+        // TODO: This is where flex-basis would come into play
+        child.Measure(availableSize);
+        winrt::Size childDesiredSize = child.DesiredSize();
+
+      //flexbasis
+    //Asign initial size based on FlexBasis
+        float flexBasis = 0;
+
+        float inputBasis = float((GetBasis(child)));
+
+        
+        if (inputBasis > 0) {
+            flexBasis = float((GetBasis(child)));
+            childDesiredSize = CreateSize(inputBasis, CrossAxis(childDesiredSize));
```

Comment:
note our logic here only handles the cases where flexDirection is 'row'. We will want to extend this logic to also handle the case where flexDirection is column and the flexBasis value will apply to the items height.